providers/dns/nearlyfreespeech/internal/client.go

```diff
@@ -90,20 +103,30 @@ func (c Client) doRequest(ctx context.Context, endpoint *url.URL, params url.Val
 	return nil
 }
 
-func (c Client) createSignature(uri string, body string) string {
-	// This is the only part of this that needs to be serialized.
+//nolint:gochecknoglobals
+var genSalt = func() []byte {
 	salt := make([]byte, 16)
 	for i := 0; i < 16; i++ {
 		salt[i] = saltBytes[rand.Intn(len(saltBytes))]
 	}
+	return salt
+}
+
+//nolint:gochecknoglobals
+var timeNow = time.Now
```

Comment:
you can use global variables.

Reply:
Should I add these as more fields in Client then, so that tests can override them?

Reply:
yes and no:
- you can modify the signature of the function `createSignature`
- you have to add tests on this function and not modify the existing tests

Reply:
I have another proposal for the `createSignature` function:
```go
type Signer struct {
	saltShaker func() []byte
	clock      func() time.Time
}

func (c Signer) Sign(uri string, body, login, apiKey string) string {
	// Header is "login;timestamp;salt;hash".
	// hash is SHA1("login;timestamp;salt;api-key;request-uri;body-hash")
	// and body-hash is SHA1(body).

	bodyHash := sha1.Sum([]byte(body))
	timestamp := strconv.FormatInt(c.clock().Unix(), 10)

	uri = "/" + strings.TrimLeft(uri, "/")

	hashInput := fmt.Sprintf("%s;%s;%s;%s;%s;%02x", login, timestamp, c.saltShaker(), apiKey, uri, bodyHash)

	return fmt.Sprintf("%s;%s;%s;%02x", login, timestamp, c.saltShaker(), sha1.Sum([]byte(hashInput)))
}

func getSalt() []byte {
	// This is the only part of this that needs to be serialized.
	salt := make([]byte, 16)
	for i := 0; i < 16; i++ {
		salt[i] = saltBytes[rand.Intn(len(saltBytes))]
	}

	return salt
}
```

Reply:
Ah, I haven't noticed your message and have done basically the same thing in `Client`. Does that seem fine?

Reply:
It's close to my proposal, but there is a difference in the testability: with my proposal, I can test the signature part in isolation.